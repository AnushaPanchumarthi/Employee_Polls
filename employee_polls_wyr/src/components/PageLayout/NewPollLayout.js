import React, { useState } from 'react';
import { connect } from 'react-redux';
import { Navigate, useNavigate ,useLocation} from 'react-router-dom';
import { handleCreateQuestion } from '../../actions/questions';

const NewPollLayout = (props) => {
  console.log('NEW POLL...', props)
  const navigate = useNavigate();
  const location = useLocation();
  const [optionOneText, setOptionOneText] = useState('');
  const [optionTwoText, setoptionTwoText] = useState('');
  const [ redirect , setredirect ] = useState(false);

  const handleOptionOneChange = (e) => {
    setOptionOneText(e.target.value);
  };
  const handleOptionTwoChange = (e) => {
    setoptionTwoText(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    const newPoll = {
      optionOneText: optionOneText,
      optionTwoText: optionTwoText,
      author: props.authedUser
    };
    props.dispatch(handleCreateQuestion(newPoll));
    setOptionOneText('');
    setoptionTwoText('');
    setredirect(true);
  };
  
  if(props.loaded === undefined){
    return <p>Loading...</p>
  }
  if(props.loaded === true && (props.authedUser === null || props.authedUser === undefined )){
    console.log('I am satisfied with not found', location.pathname);
    return  <Navigate to="/login" replace state={{ path: '/404' }} />
  }
  if( redirect === true){
    return  navigate('/');
  }

  return (
    <form onSubmit={handleSubmit} className="w-75 bg-light px-4 py-4">
      <h6>{props.authedUsername} Asked Would You Rather:</h6>
      <div className="form-group row mt-4">
        <label htmlFor="optionOne" className="col-sm-2 col-form-label">
          Option One
        </label>
        <div className="col-sm-9">
          <input
            type="text"
            name="optionOne"
            value={optionOneText}
            onChange={handleOptionOneChange}
            className="form-control"
            id="optionOne"
            required
          />
        </div>
      </div>
      <div className="form-group row mt-4">
        <label htmlFor="optionTwo" className="col-sm-2 col-form-label">
          Option Two
        </label>
        <div className="col-sm-9">
          <input
            type="text"
            name="optionTwo"
            value={optionTwoText}
            onChange={handleOptionTwoChange}
            className="form-control"
            id="optionTwo"
            required
          />
        </div>
      </div>
      <div className="text-center">
        <button
          className="btn btn-dark mt-4"
          disabled={optionOneText === undefined && optionTwoText === undefined}
          type="submit">
          Submit
        </button>
      </div>
    </form>
  );
};

const mapStateToProps = ({ authedUser, users }) => {
  return {
    loaded: true,
    authedUser,
    authedUsername: authedUser !== null ? users[authedUser].name : ''
  };
};

export default connect(mapStateToProps)(NewPollLayout);
