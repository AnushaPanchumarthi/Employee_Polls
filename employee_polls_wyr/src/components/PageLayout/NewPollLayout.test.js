import * as React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import reducer from '../../reducers';
import middleware from '../../middleware';
import { createStore } from 'redux';
import ReactDOM from 'react-dom';
import { MemoryRouter as Router } from 'react-router-dom';
import NewPollLayout from './NewPollLayout';

let root;

beforeEach(() => {
  root = document.createElement('div');
  document.body.appendChild(root);
});

afterEach(() => {
  // cleanup on exiting
  ReactDOM.unmountComponentAtNode(root);
  root.remove();
  root = null;
});

describe('NewPoll', () => {
  it('should match snapshot', () => {
    const store = createStore(reducer, middleware);
    const component = render(
      <Provider store={store}>
        <Router>
          <NewPollLayout />
        </Router>
      </Provider>,
      root
    );
    expect(component).toMatchSnapshot();
  });
});
