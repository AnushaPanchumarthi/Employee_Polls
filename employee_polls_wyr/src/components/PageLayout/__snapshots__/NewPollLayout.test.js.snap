// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <form
        class="w-75 bg-light px-4 py-4"
      >
        <h6>
           Asked Would You Rather:
        </h6>
        <div
          class="form-group row mt-4"
        >
          <label
            class="col-sm-2 col-form-label"
            for="optionOne"
          >
            Option One
          </label>
          <div
            class="col-sm-9"
          >
            <input
              class="form-control"
              id="optionOne"
              name="optionOne"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="form-group row mt-4"
        >
          <label
            class="col-sm-2 col-form-label"
            for="optionTwo"
          >
            Option Two
          </label>
          <div
            class="col-sm-9"
          >
            <input
              class="form-control"
              id="optionTwo"
              name="optionTwo"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="text-center"
        >
          <button
            class="btn btn-dark mt-4"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="w-75 bg-light px-4 py-4"
    >
      <h6>
         Asked Would You Rather:
      </h6>
      <div
        class="form-group row mt-4"
      >
        <label
          class="col-sm-2 col-form-label"
          for="optionOne"
        >
          Option One
        </label>
        <div
          class="col-sm-9"
        >
          <input
            class="form-control"
            id="optionOne"
            name="optionOne"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="form-group row mt-4"
      >
        <label
          class="col-sm-2 col-form-label"
          for="optionTwo"
        >
          Option Two
        </label>
        <div
          class="col-sm-9"
        >
          <input
            class="form-control"
            id="optionTwo"
            name="optionTwo"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="text-center"
      >
        <button
          class="btn btn-dark mt-4"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
